blog/en/blog/2022/10/27/ten-use-cases-api-gateway.md

```diff
@@ -0,0 +1,180 @@
+
+![API Caching with APISIX Api Gateway](https://static.apiseven.com/2022/10/25/6357acd141b49.jpg)
+
+You can read more about [API Gateway Caching with Apache APISIX](https://medium.com/@ApacheAPISIX/api-gateway-caching-for-asp-net-core-web-api-cf24d0e598fc) in the dedicated blog post.
+
+## API Fault handling
+
+It is critical to handle failures in service-to-service calls gracefully. Also, we want to be sure that any resilience mechanisms we have in place such as error handling code, [circuit breaker](https://dev.to/apisix/implementing-resilient-applications-with-api-gateway-circuit-breaker-ggk), [health checks](https://dev.to/apisix/implementing-resilient-applications-with-api-gateway-health-check-338c), [retry](https://docs.microsoft.com/en-us/azure/architecture/patterns/retry), fallback, redundant instances, and so on. Modern API Gateways support all the above error-handling features including automatic retries and timeouts.
+
+![API Fault handling with Apache APISIX](https://static.apiseven.com/2022/10/25/6357b63234ace.png)
+
+For example, APISIX acts as an orchestrator that can use this status report to decide how to manage the traffic, load balance to a healthy node, fail-fast due to some cascading failures or simply alerts you when it notices something goes wrong. API Gateway also ensures that routing and other network-level components work together successfully to deliver a request to the API process. It helps you to detect in the early stage and fix issues for your running application much more easily.
+
+## Sidecar API gateway
+
+**Sidecar** implements an API gateway as a container attached to a service in an independent runtime, such as [Kubernetes](https://kubernetes.io/). Sidecar is a pattern that corresponds to a [sidecar](https://en.wikipedia.org/wiki/Sidecar) attached to a motorcycle, similarly, it is attached to a parent application and provides supporting features for the application. The sidecar also shares the same lifecycle as the parent application, is created and retired alongside the parent, and introduces additional features such as monitoring, logging, configuration, and networking services. The benefits of adopting this pattern are that each service runtime can configure its own API gateway in the best way. Because the requirement to enable the API gateway functionalities and setups can vary from service to service. At the same time, it separates concerns if an issue occurs in the shared API gateway infrastructure then all services will not be impacted.
```

Comment:
It's too far-fetched IMHO. I wouldn't use it.

Reply:
Right, it is too much and maybe I might dedicated blog post for sidecar now I replaced with API Versioning and promoted your content at the same time:))

https://blog.frankel.ch/evolve-apis/